packages/jazz-tools/src/coValues/coList.ts

```diff
@@ -446,6 +449,77 @@ export class CoList<Item = any> extends Array<Item> implements CoValue {
     ): InstanceType<Cl> {
         return cl.fromRaw(this._raw) as InstanceType<Cl>;
     }
+
+    asPlainData<L extends CoList>(this: L): RecursiveCoMapInit<L>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depth: Depth & DepthsIn<L>,
+    ): Promise<RecursiveCoMapInit<L> | undefined>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depthOrSeen?:
+            | (Depth & DepthsIn<L>)
+            | WeakMap<CoMap | CoList, RecursiveCoMapInit<L>>,
+    ): RecursiveCoMapInit<L> | Promise<RecursiveCoMapInit<L> | undefined> {
+        if (depthOrSeen instanceof WeakMap) {
+            return this.asPlainDataSync(depthOrSeen);
+        } else if (depthOrSeen === undefined) {
+            return this.asPlainDataSync(new WeakMap());
+        } else {
+            return this.asPlainDataAsync(depthOrSeen, new WeakMap());
+        }
+    }
+
+    private asPlainDataSync<L extends CoList>(
+        this: L,
+        seen: WeakMap<CoMap | CoList, RecursiveCoMapInit<UnCo<L[number]>>[]>,
+    ): RecursiveCoMapInit<L> {
+        if (seen.has(this)) {
+            return seen.get(this) as RecursiveCoMapInit<L>;
+        }
+        const plainArray: Array<RecursiveCoMapInit<UnCo<L[number]>>> = [];
```

Comment:
**suggestion:** since we know the size of the array beforehand we can use `new Array(this.length)` the reduce the number of reallocations to 1 when filling the array

Can look like a micro-optimization, but on big sized arrays makes some difference.